### ISSUE TITLE
jasmine option "Stop execution on spec failure" does not work properly in combination with karma

### ISSUE THREAD

**Issue by maximedupre** [karma-runner/karma-jasmine#218]
## Environment
* Jasmine core : `v3.1.0`
* karma-jasmine: `v1.1.2`
* Chrome 69
* Operating System and version (desktop or mobile): OSX 10.13.6

Please see https://github.com/jasmine/jasmine/issues/1576

**Comment by johnjbarton** [comment]
I guess this will be fixed by #224

**Comment by DirkToewe** [comment]
Is there any update on this? `stopSpecOnExpectationFailure: true` is not working for me with `jasmine-core@3.3.0` and `karma-jasmine@2.0.1`. I've also tried `oneFailurePerSpec: true`, `throwOnExpectationFailure: true` and `failFast: true`.

**Comment by johnjbarton** [comment]
Please run karma with `--log-level debug` or the equivalent config file value. Check the 'final' config. It should contain 'clientConfig' with `oneFailurePerSpec`. That value will be sent to jasmine when the start function is created: https://github.com/karma-runner/karma-jasmine/blob/master/src/adapter.js#L366

**Comment by DirkToewe** [comment]
I had already done that ... kind of ... using `console.log` inside of karma-jasmine's createStartFn. But just to be sure here's  the debug output:
```js
// ...
{
   "random":true,
   "oneFailurePerSpec":true,
   "failFast":true,
   "timeoutInterval":3600000
}
// ...
```

I am fairly certain that the config is correct because if I change `createStartFn` from:
```js
function createStartFn (karma, jasmineEnv) {
  return function () {
    var clientConfig = karma.config || {}
    var jasmineConfig = clientConfig.jasmine || {}
    jasmineEnv = jasmineEnv || window.jasmine.getEnv()
    jasmineEnv.configure(jasmineConfig)
    window.jasmine.DEFAULT_TIMEOUT_INTERVAL = jasmineConfig.timeoutInterval ||
       window.jasmine.DEFAULT_TIMEOUT_INTERVAL
    jasmineEnv.addReporter(new KarmaReporter(karma, jasmineEnv))
    jasmineEnv.execute()
  }
}
```
to:
```js
function createStartFn (karma, jasmineEnv) {
  var clientConfig = karma.config || {}
  var jasmineConfig = clientConfig.jasmine || {}
  jasmineEnv = jasmineEnv || window.jasmine.getEnv()
  jasmineEnv.configure(jasmineConfig)
  window.jasmine.DEFAULT_TIMEOUT_INTERVAL = jasmineConfig.timeoutInterval ||
     window.jasmine.DEFAULT_TIMEOUT_INTERVAL
  return function () {
    jasmineEnv.addReporter(new KarmaReporter(karma, jasmineEnv))
    jasmineEnv.execute()
  }
}
```

The behavior is according to `oneFailurePerSpec: true`. The prior implentation of `createStartFn()` only calls `jasmineEnv.configure(jasmineConfig)` as soon as it is called not before `boot.js` is executed.

**Comment by johnjbarton** [comment]
Ok thanks.  Seems like an unfortunate design here: critical parts of the execution sequence move between two files.

I don't see anything in `boot.js` that could not be done directly in `adapter.js` more clearly.

Here is what I would do:

1. Remove `src/adapter.js`, `adapter.wrapper` and build code.
2. Place the content of boot.js in to `lib/adapter.js`.
3. Create a new function `configureJasmine(karma, jasmineEnv)` with the body of your `createStartFn()` and call it before the call to `createStartFn()`. 
4. `createStartFn()` would just be the `return` you have.

We can start with just 3 and 4.

**Comment by DirkToewe** [comment]
Okay. I'm gonna file a PR about 3 & 4 as soon as I get to it.

**Comment by johnjbarton** [comment]
I believe this issue is fixed in karma-jasmine 2.x. Please reopen if you still see it.

**Comment by maximedupre** [comment]
Please reopen as it has not been fixed.

**Comment by dickinr** [comment]
These changes were seemingly reverted as part of https://github.com/karma-runner/karma-jasmine/pull/243 but I don't see why?

**Comment by DirkToewe** [comment]
Sadly the hacky fix made in PR #231 does not work at all anymore with karma-jasmine 2.0.1
